models/idea/index.js

```diff
@@ -274,6 +274,40 @@ class Idea extends Model {
     const cursor = await this.db.query(query, params);
     return await cursor.all();
   }
+
+
+  /**
+   * Read ideas commented by specified users
+   * @param {string[]} minimum rate - minimal query rate
+   * @param {integer} offset - pagination offset
+   * @param {integer} limit - pagination limit
+   * @returns {Promise<Idea[]>} - list of found ideas
+   */
+  static async findHighlyRated(rateBottomLimit, { offset, limit }) {
+    const query = `
+      FOR idea IN ideas
+        FOR vote IN votes
```

Comment:
Another way of writing this may be

```
FOR idea IN ideas
  LET ideaVotes = (FOR v, e IN 1..1 INBOUND idea, INBOUND votes RETURN e) // graph traversal and assigning to ideaVotes variable
  LET voteSum = SUM(ideaVotes[*].value) // simplifying aggregation
  ...
```

I'd really like to understand which of these would have better performance, if any difference. Now I simply don't know.

This feels a bit more efficient to me, but it's hard to judge without the internal knowledge of Arango. Perhaps we could ask on stackoverflow.

Reply:
So there is a need for performance tests - I will happily test it. Also may ask the question on the stack - after tests. Probably also you can use 'explain' for AQL and this may give some information.

Reply:
Ah, I also believe the current query will filter out all the ideas which haven't received any votes. So at least a subset of my proposed changes should be applied.

i.e. not `FOR idea IN ideas FOR vote IN votes...` but `FOR idea IN ideas LET ideaVotes=(*subquery*)...`

Reply:
'..depends on the one hand which queries you want to perform and the amount of data you have. The general rule of thumb is, that if you do not know the depth of you search, then a graph traversals are likely to be faster than joins. And if you have a known or fixed depth joins are normally faster.' from Arango slack regarding JOINS vs. GRAPHS

And this 'LEFT JOIN' case can be actualy easier to do with graph, I will look at it.